### PR TITLE
87 sponsor profile

### LINKED ISSUE #87: Sponsor profile
1) Створити можливість реєстраціЇ для спонсора. Додати при реєстрації ще одне поле, в якому буде можливість вибрати роль (або талант, або спонсор)
2) Реалізувати профіль спонсору (такий самий як і профіль таланту, але в блоці BIO створити просто кількість кудосів)
3) Зробити так, щоб картка профілю спонсора не відображалася в TalentList

### PR DESCRIPTION
**Pull request by daniilievtukhov**
closed #87

### CODE REVIEW

File: src/components/Layout/Header/Header.jsx [hunk 1]
```diff
@@ -19,14 +19,22 @@ export function Header() {
...
                 });
+        } else if (user.role === "SPONSOR") {
+            TalentsService.getSponsor(user.id, token)
+                .then((response) => {
+                    setUserInfo(response);
+                })
+                .catch((error) => {
+                    console.log(error);
+                });
         }
```
Comment (whoisalex1W) [inline src/components/Layout/Header/Header.jsx]: If in the if-statement user.id will be undefined, else if-statement will work even with empty user.id, so add user.id check in else-if

File: src/components/SponsorPage/SponsorPage.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    useEffect(() => {
+        if (editting) {
+            navigate(`${location.pathname}${location.search}#edit`, {
+                replace: true,
+            });
+        } else if (!editting) {
+            navigate(`${location.pathname}${location.search}`, {
+                replace: true,
+            });
+        }
```
Comment (whoisalex1W) [inline src/components/SponsorPage/SponsorPage.jsx]: Too long , rewrite it like:
const path = `${location.pathname}${location.search}`;
const hash = editting ? '#edit' : '';
navigate(`${path}${hash}`, { replace: true });

File: src/components/Layout/AuthModal/components/Register/Register.jsx [hunk 3]
```diff
@@ -282,30 +295,69 @@ export function Register({ switcher }) {
...
                 specialization: specialization.spec,
             };
-            register(newUser)
-                .then((response) => {
-                    resetAll();
-                    const { token, ...user } = response;
-                    setCookie("token", token, {
-                        path: "/",
-                        maxAge: 3600,
+            if (!isSponsor) {
```
Comment (whoisalex1W) [inline src/components/Layout/AuthModal/components/Register/Register.jsx]: Fix this if-else, i will send you in telegram

File: src/App/App.jsx [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { Layout } from "../components/Layout";
 import { TalentPage } from "../components/TalentPage";
 import { TalentsListPage } from "../components/TalentsListPage";
 import { ProfilePage } from "../components/ProfilePage";
+import { SponsorPage } from "../components/SponsorPage/SponsorPage";
```
Comment (Ruslanchik01) [inline src/App/App.jsx]: don't forget to delete unnecessary path

File: src/components/Layout/AuthModal/components/Register/Register.jsx [hunk 5]
```diff
@@ -225,10 +233,15 @@ export function Register({ switcher }) {
...
+        let validateSp = " ";
+        if (!isSponsor) {
+            validateSp = validateSpecialization();
+        }
+
         return (
             validateFirstName() &&
             validateLastName() &&
-            validateSpecialization() &&
+            validateSp &&
```
Comment (Ruslanchik01) [inline src/components/Layout/AuthModal/components/Register/Register.jsx]: You can delete rows 236-239
```suggestion
            (isSponsor || validateSpecialization()) && 
```

File: src/components/ListProofsPage/ListProofsPage.jsx [hunk 6]
```diff
@@ -102,7 +102,7 @@ export function ListProofsPage() {
                             text={el.text}
                             created={el.created}
                             status={el.status}
-                            myProofs={[...talentsProofs]}
+                            //   myProofs={[...talentsProofs]}
```
Comment (Ruslanchik01) [inline src/components/ListProofsPage/ListProofsPage.jsx]: You can remove this row, it's from stage 3.1

**Review by Ruslanchik01 — CHANGES_REQUESTED** [review]

**Review by Ruslanchik01 — APPROVED** [review]

**Review by whoisalex1W — APPROVED** [review]

**Pull request merged**